pylint/checkers/exceptions.py

```diff
@@ -333,19 +334,39 @@ def _check_raise_missing_from(self, node: nodes.Raise) -> None:
         # like `exc.with_traceback(whatever)`. We won't analyze these, we'll just assume
         # there's a violation on two simple cases: `raise SomeException(whatever)` and `raise
         # SomeException`.
+
         if containing_except_node.name is None:
             # The `except` doesn't have an `as exception:` part, meaning there's no way that
             # the `raise` is raising the same exception.
-            self.add_message("raise-missing-from", node=node)
-        elif isinstance(node.exc, nodes.Call) and isinstance(node.exc.func, nodes.Name):
-            # We have a `raise SomeException(whatever)`.
-            self.add_message("raise-missing-from", node=node)
+            class_of_old_error = "Exception"
+            if isinstance(containing_except_node.type, nodes.Name):
+                class_of_old_error = containing_except_node.type.name
+            elif isinstance(containing_except_node.type, nodes.Tuple):
+                class_of_old_error = ", ".join(
+                    n.name for n in containing_except_node.type.elts
+                )
+            except_as_exc = f"'except {class_of_old_error} as exc' and "
+            self.add_message(
+                "raise-missing-from",
+                node=node,
+                args=(except_as_exc, node.as_string(), "exc"),
+                confidence=HIGH,
+            )
         elif (
-            isinstance(node.exc, nodes.Name)
-            and node.exc.name != containing_except_node.name.name
+            isinstance(node.exc, nodes.Call)
+            and isinstance(node.exc.func, nodes.Name)
+            or (
+                isinstance(node.exc, nodes.Name)
+                and node.exc.name != containing_except_node.name.name
+            )
```

Comment:
```suggestion
            or isinstance(node.exc, nodes.Name)
            and node.exc.name != containing_except_node.name.name
```

`And` should bind before `or` right?